mypy/messages.py

```diff
@@ -740,12 +740,16 @@ def could_not_infer_type_arguments(self, callee_type: CallableType, n: int,
     def invalid_var_arg(self, typ: Type, context: Context) -> None:
         self.fail('List or tuple expected as variable arguments', context)
 
-    def invalid_keyword_var_arg(self, typ: Type, context: Context) -> None:
-        if isinstance(typ, Instance) and (typ.type.fullname() == 'builtins.dict'):
+    def invalid_keyword_var_arg(self, typ: Type, is_mapping: bool, context: Context) -> None:
+        if isinstance(typ, Instance) and is_mapping:
             self.fail('Keywords must be strings', context)
         else:
-            self.fail('Argument after ** must be a dictionary',
-                      context)
+            suffix = ''
+            if isinstance(typ, Instance):
+                suffix = ', not {}'.format(typ.type.fullname())
+            self.fail(
+                'Argument after ** must be a subtype of Mapping{}'.format(suffix),
```

Comment:
I would just say ``must be a mapping`` (with small ``m``), not ``must be a subtype of Mapping``.